queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/name_request.py

```diff
@@ -21,13 +21,18 @@
 from flask import current_app
 from legal_api.models import Business, Filing, RegistrationBootstrap
 from legal_api.services.bootstrap import AccountService
+from legal_api.services.utils import get_str
 
 
-def consume_nr(business: Business, filing: Filing):
+def consume_nr(business: Business, filing: Filing, nr_num_path=None):
     """Update the nr to a consumed state."""
     try:
         # use the fact that getting a non-existant NR will fail with a KeyError to bail early
-        nr_num = filing.filing_json['filing']['incorporationApplication']['nameRequest']['nrNumber']
+        nr_num = ''
+        if nr_num_path:
+            nr_num = get_str(filing.filing_json, nr_num_path)
+        else:
+            nr_num = filing.filing_json['filing']['incorporationApplication']['nameRequest']['nrNumber']
```

Comment:
Since  you already have a nr_num_path added as an argument, its better to pass the path for IA as well or have that as default